### PR TITLE
Ports painting improvements, admin painting managers, and some tgui elements

### PR DESCRIPTION
**Pull request by KoboldCommando**
# About The Pull Request

This is a port of https://github.com/tgstation/tgstation/pull/63170

- Paintings can now do stroke painting.
- Added painting management panel for admins.
- Paintings now display author's character name, year of painting, medium and patron when hung on wall.
- You can become new patron by paying more than the previous one.
- Added painter's palettes to library vendor. (Sprites by @Mickyan )

In addition, this adds a few items to the library vendor (based on tg's list).
Along the way I also ported some tgui elements from tg.  Namely, tgui_alert, tgui_input_text, and tgui_input_number.  I did not apply them to other cases where they could be used, but they should be ready to go.

## Why It's Good For The Game

- FAR less tedious painting will result in many more paintings!
- Painting credit is very nice
- We've had trouble administrating paintings and this should help
- Patronage should help slightly with all that useless money
- The new tgui popups are very nice and should now be easier to apply elsewhere

## Changelog

:cl: AnturK (ported by KoboldCommando)
add: Painting now supports strokes!  No more pixel by pixel clicking.
add: The library vendor has new items, including a painting palette to help with paintings!
add: You can now sign your paintings, and they will also display the time painted and medium used
add: You can now donate money to become a patron of a painting!  Or outbid the previous patron!
code: Ported some tgui elements
admin: There is now an admin button to manage paintings
/:cl:

### CODE REVIEW

**Comment by dwasint** [comment]
how much of the tgui alert/ input list does this port? up to the modern prs or only the first prs?

File: code/controllers/subsystem/persistent_paintings.dm [hunk 1]
```diff
@@ -0,0 +1,203 @@
+#define PAINTINGS_DATA_FORMAT_VERSION 1
+/*
```
Comment (dwasint) [inline code/controllers/subsystem/persistent_paintings.dm:2]: is this comment block an example of a stored painting? i'd add a comment clarifying that

**Comment by KoboldCommando** [comment]
> how much of the tgui alert/ input list does this port? up to the modern prs or only the first prs?

It only ports just enough to make these features work.  You can see the files I brought over.  They're the latest versions though.

**Review by nednaZ — CHANGES_REQUESTED** [review]
I need the other ported PRs added to the description/credits as well, unless the TGUI changes were somehow a part of this.

Everything seems fine to me, but we'll need to testmerge this.

File: code/game/objects/structures/artstuff.dm [hunk 2]
```diff
@@ -1,462 +0,0 @@
-
```
[the remark this answers is not in the document]
Comment (KoboldCommando) [reply code/game/objects/structures/artstuff.dm]: I... thought it had been.  That's weird.  Fixed!

File: code/modules/art/paintings.dm [hunk 3]
```diff
@@ -86,63 +94,105 @@
...
+		return
+	if(offer_amount <= 0 || sniped_amount != painting_metadata.credit_value || offer_amount < painting_metadata.credit_value+1 || !user.canUseTopic(src))
+		return
+	if(!account.adjust_money(-offer_amount))
+		to_chat(user,span_warning("Transaction failure. Please try again."))
+		return
+	painting_metadata.patron_ckey = user.ckey
+	painting_metadata.patron_name = user.real_name
+	painting_metadata.credit_value = offer_amount
+	to_chat(user,span_notice("Nanotrasen Trust Foundation thanks you for your contribution. You're now offical patron of this painting."))
```
Comment (nednaZ) [inline code/modules/art/paintings.dm:194]: Grammar and spelling.

```suggestion
	to_chat(user,span_notice("Nanotrasen Trust Foundation thanks you for your contribution. You're now the official patron of this painting."))
```.

**Comment by KoboldCommando** [comment]
Fixed it so canvasses obtain their painting name immediately, rather than having to wait until they're hung.

**Comment by nednaZ** [comment]
No issues found during testmerge.
Go buy a painting.

**Pull request merged**